reactor-netty-core/src/main/java/reactor/netty/tcp/TcpServerConfig.java

```diff
@@ -122,12 +122,16 @@ protected ChannelPipelineConfigurer defaultOnChannelInit() {
 	 */
 	static final boolean SSL_DEBUG = Boolean.parseBoolean(System.getProperty(ReactorNetty.SSL_SERVER_DEBUG, "false"));
 
-	static final class MicrometerTcpServerMetricsRecorder extends MicrometerChannelMetricsRecorder {
+	public static class MicrometerTcpServerMetricsRecorder extends MicrometerChannelMetricsRecorder {
 
 		static final MicrometerTcpServerMetricsRecorder INSTANCE = new MicrometerTcpServerMetricsRecorder();
 
-		MicrometerTcpServerMetricsRecorder() {
-			super(reactor.netty.Metrics.TCP_SERVER_PREFIX, "tcp");
+		public MicrometerTcpServerMetricsRecorder(String prefix) {
+			super(prefix, "tcp");
+		}
+
+		public MicrometerTcpServerMetricsRecorder() {
+			this(reactor.netty.Metrics.TCP_SERVER_PREFIX);
```

Comment:
Is there a value of making this public? One can directly use `MicrometerChannelMetricsRecorder`.

Reply:
Yes. The reason is the same for why we created a class `MicrometerTcpServerMetricsRecorder`, that's for abstraction.
If someday the implementation of `MicrometerTcpServerMetricsRecorder` changes, users don't need to know it because they can extend `MicrometerTcpServerMetricsRecorder`.